src/allotropy/parsers/roche_cedex_hires/roche_cedex_hires_parser.py

```diff
@@ -0,0 +1,437 @@
+                        ),
+                        measurement_identifier=random_uuid_str(),
+                        sample_document=add_custom_information_document(
+                            SampleDocument(
+                                sample_identifier=get_value_not_none(
+                                    data, "Sample identifer", row
+                                ),
+                                batch_identifier=get_value(
+                                    data, "Reactor identifer", row
+                                ),
+                            ),
+                            sample_custom_document,
+                        ),
+                        device_control_aggregate_document=CellCountingDetectorDeviceControlAggregateDocument(
+                            device_control_document=[
+                                DeviceControlDocumentItemModel(
+                                    device_type=constants.DEVICE_TYPE,
+                                    detection_type=constants.DETECTION_TYPE,
+                                    sample_volume_setting=get_property_value(
+                                        data,
+                                        "Sample volume",
+                                        row,
+                                        TQuantityValueMicroliter,
+                                    ),
+                                )
+                            ]
+                        ),
+                        processed_data_aggregate_document=ProcessedDataAggregateDocument1(
+                            processed_data_document=[
+                                add_custom_information_document(
+                                    ProcessedDataDocumentItem(
+                                        data_processing_document=DataProcessingDocument(
+                                            cell_type_processing_method=get_value(
+                                                data, "Cell type name", row
+                                            ),
+                                            cell_density_dilution_factor=get_property_value(
+                                                data,
+                                                "Dilution",
+                                                row,
+                                                TQuantityValueUnitless,
+                                            ),
+                                        ),
+                                        viability__cell_counter_=TQuantityValuePercent(
+                                            value=get_value_not_none(
+                                                data, "Viability", row
+                                            )
+                                        ),
+                                        viable_cell_density__cell_counter_=TQuantityValueMillionCellsPerMilliliter(
+                                            value=float(
```

Comment:
Use `get_calculated_data`?

Reply:
You can handle this by wrapping `get_calculated_data` in `assert_not_none` from `allotropy.parsers.utils.values`

Reply:
Fixed this!